lib/babe/errors.go

```diff
@@ -86,13 +86,33 @@ func (e DispatchOutcomeError) Error() string {
 
 // A TransactionValidityError is possible errors while checking the validity of a transaction
 type TransactionValidityError struct {
-	msg string // description of error
+	msg transactionValidityErrorMsg // description of error
 }
 
 func (e TransactionValidityError) Error() string {
 	return fmt.Sprintf("transaction validity error: %s", e.msg)
 }
 
+type transactionValidityErrorMsg string
```

Comment:
why this type is necessary?

Reply:
so that not any string can be used as the error message, only the predefined ones. I can remove it if you think it's overkill

Reply:
Got it! IMO seems overkill, I think that each of the errors string var could be of the type `error` like that: 
```go
var (
  ErrunexpectedTxCallMsg  = errors.New("call of the transaction is not expected")
  ErrinvalidPaymentMsg  = errors.New("invalid payment")
  ...
)
```
then, will be easy to determine a if an error occurs:
```go
err := determineErrType(...)
if errors.Is(err, ErrunexpectedTxCallMsg) { ... }
```

Reply:
updated the errors!